staging/src/k8s.io/apiserver/pkg/storage/interfaces.go

```diff
@@ -184,26 +184,23 @@ type Interface interface {
 	// (e.g. reconnecting without missing any updates).
 	// If resource version is "0", this interface will list current objects directory defined by key
 	// and send them in "ADDED" events, before watch starts.
-	WatchList(ctx context.Context, key string, resourceVersion string, p SelectionPredicate) (watch.Interface, error)
+	WatchList(ctx context.Context, key string, opts ListOptions) (watch.Interface, error)
 
 	// Get unmarshals json found at key into objPtr. On a not found error, will either
-	// return a zero object of the requested type, or an error, depending on ignoreNotFound.
+	// return a zero object of the requested type, or an error, depending on 'opts.ignoreNotFound'.
 	// Treats empty responses and nil response nodes exactly like a not found error.
-	// The returned contents may be delayed, but it is guaranteed that they will
-	// be have at least 'resourceVersion'.
-	Get(ctx context.Context, key string, resourceVersion string, objPtr runtime.Object, ignoreNotFound bool) error
+	// The returned contents may be delayed.
```

Comment:
I agree that the previous comment wasn't ideal, but now it's too vague imho - then may be delayed only for RV!="" and the minRV semantic is guaranteed. Can you please update the comment?

Reply:
Same below

Reply:
Actually, maybe what we need is:

... delayed according to the semantic of GetOptions.ResourceVersion.

or sth like that

Reply:
Thanks, looks like this got messed up when I split this PR out from https://github.com/kubernetes/kubernetes/pull/91505. I'll add "delayed according to the semantic of GetOptions.ResourceVersion" for now and then merge in some more detail in https://github.com/kubernetes/kubernetes/pull/91505.